scripts/apps/search/directives/SearchContainer.ts

```diff
@@ -12,8 +12,20 @@ export function SearchContainer() {
                 pageTitle.setUrl(gettext('Search'));
 
                 $scope.$watch('search.activeProvider', (activeProvider) => {
-                    if (activeProvider && activeProvider.advanced_search !== undefined) {
-                        this.flags.facets = !!activeProvider.advanced_search;
+                    if (activeProvider) {
+                        if (activeProvider.advanced_search !== undefined) {
+                            this.flags.facets = !!activeProvider.advanced_search;
+                        }
+                        if (activeProvider.config?.default_list_view) {
+                            switch (activeProvider.config.default_list_view) {
+                            case 'list':
+                                $scope.view = 'compact';
```

Comment:
what about `$scope.view = activeProvider.config.default_list_view;` and use `compact/photogrid` as `viewId`

Reply:
I thought about it, but that would make the UI and the DB very coupled together. I mean that this naming is used in the search view but also in other places. If we wanna change the UI we would change the data too instead of the code only. What do you think?

Reply:
I agree with Pablo on this. It's easier when all possible js variable values can be found statically from the code. It's also easier to rename and is clear what has to be changed in DB if necessary.